lib/src/clojure_lsp/feature/diagnostics.clj

```diff
@@ -193,49 +198,28 @@
          (map (fn [unused-var]
                 (unused-public-var->finding unused-var kondo-config))))))
 
-(defn ^:private file-var-definitions [project-db filename]
-  (q/find-var-definitions project-db filename false))
-(def ^:private file-kw-definitions q/find-keyword-definitions)
-(def ^:private all-var-definitions q/find-all-var-definitions)
-(def ^:private all-kw-definitions q/find-all-keyword-definitions)
-
-(defn project-findings
+(defn ^:private findings-of-project
   [db kondo-config]
   (let [project-db (q/db-with-internal-analysis db)]
-    (unused-public-vars (all-var-definitions project-db)
-                        (all-kw-definitions project-db)
-                        project-db kondo-config)))
+    (unused-public-vars project-db project-db kondo-config)))
 
-(defn files-findings
-  [filenames db kondo-config]
+(defn ^:private findings-of-uris
+  [uris db kondo-config]
   (let [project-db (q/db-with-internal-analysis db)
-        files-db (update project-db :analysis select-keys filenames)]
-    (unused-public-vars (all-var-definitions files-db)
-                        (all-kw-definitions files-db)
-                        project-db kondo-config)))
-
-(defn file-findings
-  [filename db kondo-config]
-  (let [project-db (q/db-with-internal-analysis db)]
-    (unused-public-vars (file-var-definitions project-db filename)
-                        (file-kw-definitions project-db filename)
-                        project-db kondo-config)))
+        db-of-uris (update project-db :analysis select-keys uris)]
+    (unused-public-vars db-of-uris project-db kondo-config)))
 
 (defn ^:private finalize-findings! [findings reg-finding!]
-  (run! reg-finding! findings)
-  (group-by :filename findings))
+  (let [uri->filename (memoize shared/uri->filename)]
+    (run! #(reg-finding! (assoc % :filename (uri->filename (:uri %)))) findings))
```

Comment:
If kondo support uri internally in the future we could remove this in the future

Reply:
Yep, that's true. I think `clojure-lsp.feature.diagnostics` and `clojure-lsp.kondo` will need some changes when that happens, but hopefully not nearly as many other places.